src/schema/v2/users/updateUserMutation.ts

```diff
@@ -37,24 +38,35 @@ export const updateUserMutation = mutationWithClientMutationId<
     email: { type: GraphQLString },
     name: { type: GraphQLString },
     phone: { type: GraphQLString },
+    adminNote: { type: GraphQLString },
   },
   outputFields: {},
-  mutateAndGetPayload: async (args, { updateUserLoader }) => {
-    if (!updateUserLoader) {
+  mutateAndGetPayload: async (
+    args,
+    { updateUserLoader, updateAdminNotesLoader }
+  ) => {
+    if (!updateUserLoader || !updateAdminNotesLoader) {
       throw new Error(
         "You need to be signed in as an admin to perform this action"
       )
     }
 
-    const gravityOptions = Object.keys(args)
-      .filter((key) => key !== "id")
+    const updateUserLoaderPayload = Object.keys(args)
+      .filter((key) => key !== "id" && key !== "adminNote")
       .reduce(
         (acc, key) => ({ ...acc, [snakeCase(key)]: args[key] }),
         {} as GravityInput
       )
 
+    const updateAdminNotesLoaderPayload = {
+      body: args.adminNote,
+    }
+
     try {
-      return await updateUserLoader?.(args.id, gravityOptions)
+      if (args.adminNote) {
+        await updateAdminNotesLoader(args.id, updateAdminNotesLoaderPayload)
```

Comment:
combo'ing several updates here into one more general mutation is interesting...what if one of them fails, is there a complex error-handling state for the client? It _might_ be somewhat verbose, but nevertheless boiler-plate-y and ultimately easier to add a new `createAdminNoteMutation`.

Reply:
Hmm. My thinking is in the first iteration, it would be sufficient to display the error thrown in the `catch` block on client side, so not clear why the complex error handling would be necessary? That being said, I hear you on the approach here probably not being the most consistent with the rest of MP, and it is probably better to decouple the two mutations.

Reply:
I mean that let's say the admin note call succeeds, and then the update user one fails. If the user tries again (maybe after updating some error'ing field, or pressing a button again, or there's a retry of some kind), they might start creating redundant admin notes.

Reply:
Yeah. More than fair point. Should have considered idempotency required here. I've broken it out into its own mutation.